### ISSUE TITLE
KNX Telegrams will not be received anymore

### ISSUE THREAD

**Issue by ChriD** [iobroker-community-adapters/ioBroker.openknx#330]
Hi,
After a while (timespan has to be analyzed yet) the adapter will not receive any knx telegrams or will not update the state for the knx ga (not clear yet).  I started 3 days ago with testing and every day i had to restart the adapter again after a while to work correctly.
I'll investigate it further and set the log level to debug

The last lines in the log for openknx were following
```
2023-01-10 13:40:18.299 - warn: openknx.0 (980024) Connection lost
2023-01-10 13:40:18.311 - warn: openknx.0 (980024) timed out waiting for CONNECTIONSTATE_RESPONSE
2023-01-10 13:40:18.340 - info: openknx.0 (980024) Connected!
```
After that there were no entries until i restarted the adapter at about 18:00

**Versions:**  
 - Adapter version: 0.5.2

**Comment by ChriD** [comment]
Today at 8:17 the same thing happend again. The last entries in the log were:
```
2023-01-11 08:16:45.584 - debug: openknx.0 (1064443) Inbound GroupValue_Write from 1.1.171 GA 7/2/1 to Object: openknx.0.Umweltsensoren.Feuchtigkeit.Taster_\|_Küche_\|_Hebeschiebe_\|_Feuchtigkeit val: 39.36 dpt: DPT9.007
2023-01-11 08:16:45.608 - debug: openknx.0 (1064443) Inbound GroupValue_Write from 1.1.171 GA 7/3/1 to Object: openknx.0.Umweltsensoren.Druck.Taster_\|_Küche_\|_Hebeschiebe_\|_Druck val: 945.4503173828125 dpt: DPT14.058
```

Nothing else. No connection lost or connected like in the above post. This is the last Telegram received from the bus. Then there was absolute silence until 9:30 (That can't be on the bus because i have things that write every 5 minutes . 
But then following log appeard

```
2023-01-11 09:30:21.784 - warn: openknx.0 (1064443) Connection lost
2023-01-11 09:30:21.802 - warn: openknx.0 (1064443) timed out waiting for CONNECTIONSTATE_RESPONSE
2023-01-11 09:30:21.831 - info: openknx.0 (1064443) Connected!
2023-01-11 09:30:24.970 - debug: openknx.0 (1064443) Inbound GroupValue_Write from 1.1.171 GA 7/1/1 to Object: openknx.0.Umweltsensoren.Temperatur.Taster_\|_Küche_\|_Hebeschiebe_\|_Temperatur 
```
and system retrieved telegrams again.
I have a diy home automation system which i am replacing with iobroker. There i use the
https://bitbucket.org/ekarak/knx.js/src/master/ knx stack and this does not have the problem. So it should not be on the hardware side.

**Comment by boellner** [comment]
Hi, 
-did you try/experience this  with older adapter versions?
-what hardware setup are you using?
-are you able to do a ethernet trace with wireshark at the time of the problem or even in the transition phases?

**Comment by ChriD** [comment]
> did you try/experience this with older adapter versions?

I did not try older versions. But maybe i will.

> what hardware setup are you using

Odroid C4 with DietPi. Currently it is connected via WLAN, I'll try a wired connection next

> are you able to do a ethernet trace with wireshark at the time of the problem or even in the transition phases?

I think if the problem persists i'll create a simple test adapter with the knx lib i use in my DIY-System. If this lib will behave the same way the problem is not related to the knx stack you use (But i think you use the same lib as i did, bur maybe another version)
Catching this with wireshark may be a bit of a pain

**Comment by boellner** [comment]
I did some modifications in the knx library, but not the part with the connection. 

> Odroid C4 with DietPi. Currently it is connected via WLAN, I'll try a wired connection next

I mean what type of IP to KNX interface do you use, e.g. the KNX IP Interface 731 - Weinzierl Engineering GmbH
If your problems comes back, it would be interesting to see what this device does (traffic leds, ethernet,..)

**Comment by ChriD** [comment]
i switched The Odroid C4 network connection from WLAN to LAN and it seems to work better now. I will check this for a view days and will give you info.  
Not sure why there are KNX connection problems when using WLAN while the other stuff seems to work. The Odroid WLAN traffic runs fine if the problem appears, and the Artnet-DMX adapter works too.
If it will run stable in LAN mode i will close this issue because in "production usage" i will use LAN anyway.

**Comment by ChriD** [comment]
I use the adapter now for about a week in production. In this week the adapter failed twice.
But it seems to be another problem due it didnt recover itself after a view hours. I had to restart the adapter in those cases to receive telegrams again. Nothing in the logs

**Comment by ChriD** [comment]
Today at  8am it happend again. No telegrams where received anymore.
Adapter is running 

![image](https://user-images.githubusercontent.com/2505067/219072483-52987067-93d3-4f07-8cd8-f8c9af015b50.png)

and last debug message was:

`2023-02-15 08:10:29.607 - debug: openknx.0 (63681) Inbound GroupValue_Write from 1.1.155 GA 7/3/25 to Object: openknx.0.Umweltsensoren.Druck.Taster_|_Kizi_West_|_Eingang_|_Druck val: 953.6295166015625 dpt: DPT14.058`

No crash, no disconnect info.... nothing....

**Comment by ChriD** [comment]
I have now made a script which will restart the adapter if i do not get any knx telegram within 3 minutes. (I get temperature telegrams from switches every minute, so its a good indicator if the adapter is stalling). It's the only thing i can for now. As i can see there are a lot of reboots.... Today tehere were 7.

![image](https://user-images.githubusercontent.com/2505067/221430056-1c8b1d41-b9c2-4949-bf84-f8d9bea80f2f.png)

**Comment by axelrathey** [comment]
0.5.2 became stable from version 0.2.5 already a few days ago. All the version between never were stable declared.
Formerly stable version was 0.2.5.

As I had similar problems too after updating to 0.5.2, I manually installed the old 0.2.5 from guthub via expert mode in admin.

https://github.com/iobroker-community-adapters/ioBroker.openknx/archive/refs/tags/v0.2.5.tar.gz

After this openknx version 0.2.5 works again.

I then tried version 0.2.6 but it does not work. Again I went back to version 0.2.5

Axel
